src/datastore/postgres-store.ts

```diff
@@ -1714,9 +1718,11 @@ export class PgDataStore
     // Update `principal_stx_txs`
     await client.query(
       `UPDATE principal_stx_txs
-      SET canonical = $1, microblock_canonical = $2
-      WHERE tx_id = ANY ($3)`,
-      [args.isCanonical, args.isMicroCanonical, updatedMbTxs.map(tx => hexToBuffer(tx.tx_id))]
+      SET microblock_canonical = $1, canonical = $2, index_block_hash = $3
```

Comment:
Have you performed a full (archival) event-replay with this PR? I'm wondering if the `principal_stx_txs` unique constraint might get violated during the re-org handling around here.

Reply:
Just performed one, and it was all good 👍